ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_repo-layout/settings.tsx

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useSuspenseQuery } from '@tanstack/react-query';
+import { createFileRoute, useNavigate } from '@tanstack/react-router';
+import { Loader2 } from 'lucide-react';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import {
+  useRepositoriesServiceDeleteRepositoryById,
+  useRepositoriesServiceGetRepositoryById,
+  useRepositoriesServiceGetRepositoryByIdKey,
+  useRepositoriesServicePatchRepositoryById,
+} from '@/api/queries';
+import { $RepositoryType, ApiError, RepositoriesService } from '@/api/requests';
+import { DeleteDialog } from '@/components/delete-dialog';
+import { ToastError } from '@/components/toast-error';
+import { Button } from '@/components/ui/button';
+import {
+  Card,
+  CardContent,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form';
+import { Input } from '@/components/ui/input';
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import { toast } from '@/lib/toast';
+
+const formSchema = z.object({
+  url: z.string(),
+  type: z.enum($RepositoryType.enum),
+});
+
+const RepositorySettingsPage = () => {
+  const params = Route.useParams();
+  const navigate = useNavigate();
+
+  const { data: repository } = useSuspenseQuery({
+    queryKey: [
+      useRepositoriesServiceGetRepositoryById,
+      params.orgId,
+      params.productId,
+      params.repoId,
+    ],
+    queryFn: async () =>
+      await RepositoriesService.getRepositoryById({
+        repositoryId: Number.parseInt(params.repoId),
+      }),
+  });
```

Comment:
Is there a particular reason why a suspense hook is not used here? Also, can you explain why there are extra elements in the query key, which are not parameters of the query?

Reply:
No particular reason, as the settings page is just renamed [edit page](https://github.com/eclipse-apoapsis/ort-server/blame/main/ui/src/routes/_layout/organizations/%24orgId/products/%24productId/repositories/%24repoId/_repo-layout/edit.tsx) with the addition of the danger zone including a delete button.